epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyVisibilityTracker.java

```diff
@@ -58,10 +65,46 @@
 
   private boolean onChangedEnabled = true;
 
+  /** All nested visibility trackers */
+  private Map<RecyclerView, EpoxyVisibilityTracker> nestedTrackers = new HashMap<>();
+
   /** This flag is for optimizing the process on detach. If detach is from data changed then it
    * need to re-process all views, else no need (ex: scroll). */
   private boolean visibleDataChanged = false;
 
+  /**
+   * Register a nested recycler view. If the parent is a recycler view with a visibility tracker
+   * attached the visibility tracking will be enabled on the nested. The parent will also notify
+   * the nested tracker if necessary.
+   *
+   * You are responsible to call this method if you are not using
+   * {@link com.airbnb.epoxy.EpoxyRecyclerView}
+   *
+   * @param nestedRecyclerView the nested recycler view
+   * @return a visibility tracker if one was attached
+   */
+  @Nullable
+  public static EpoxyVisibilityTracker registerNestedRecyclerView(
+      @NonNull RecyclerView parent,
+      @NonNull RecyclerView nestedRecyclerView
+  ) {
+    EpoxyVisibilityTracker nestedTracker = ATTACHED_TRACKERS.get(nestedRecyclerView);
+    if (nestedRecyclerView.getAdapter() instanceof BaseEpoxyAdapter) {
```

Comment:
should we instead enforce this in `attach` to catch errors earlier upstream?

in general I'm not too comfortable using the adapter to check whether the recyclerview is valid, because it's possible to have a null adapter and have the adapter set later.